src/routes/index.svelte

```diff
@@ -1,19 +1,33 @@
 <script lang="ts">
-	import MissionCard from '$lib/MissionCard.svelte';
+	import MissionCard from '$lib/cards/MissionCard.svelte';
+	import HomeSearchBar from '$lib/comp/HomeSearchBar.svelte';
 	import type { Mission } from '$lib/types';
+	import { onMount } from 'svelte';
+
 
 	export let missions: Mission[];
+	export let missionCards: any = {};
 
 	missions.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
+
+	onMount(async ()=> {
+		for (let i = 0; i < missions.length; i++)
+			missions[i].id = "mission-" + i;
```

Comment:
I don't think that this is used and the `id` field conflicts with `ID<Mission>`.